std/shared/src/main/scala/cats/effect/std/Dispatcher.scala

```diff
@@ -68,28 +68,28 @@ trait Dispatcher[F[_]] extends DispatcherPlatform[F] {
   /**
    * Submits an effect to be executed with fire-and-forget semantics.
    */
-  def unsafeRunAndForget[A](fa: F[A]): Unit = {
-    unsafeRunAsync(fa) {
-      case Left(t) => t.printStackTrace()
-      case Right(_) => ()
-    }
-  }
+  def unsafeRunAndForget[A](fa: F[A]): Unit =
+    unsafeToFutureCancelable(fa)
+      ._1
+      .onComplete {
+        case Failure(ex) => ex.printStackTrace()
+        case _ => ()
+      }(parasiticEC)
 
   // package-private because it's just an internal utility which supports specific implementations
   // anyone who needs this type of thing should use unsafeToFuture and then onComplete
-  private[std] def unsafeRunAsync[A](fa: F[A])(cb: Either[Throwable, A] => Unit): Unit = {
-    // this is safe because the only invocation will be cb
-    implicit val parasitic: ExecutionContext = new ExecutionContext {
-      def execute(runnable: Runnable) = runnable.run()
-      def reportFailure(t: Throwable) = t.printStackTrace()
-    }
-
-    unsafeToFuture(fa).onComplete(t => cb(t.toEither))
-  }
+  private[std] def unsafeRunAsync[A](fa: F[A])(cb: Either[Throwable, A] => Unit): Unit =
+    unsafeToFutureCancelable(fa)._1.onComplete(t => cb(t.toEither))(parasiticEC)
```

Comment:
Isn't this just inlining `unsafeToFuture`?

```suggestion
    unsafeToFuture(fa).onComplete(t => cb(t.toEither))(parasiticEC)
```

Reply:
It is. If the speed up is negligible, let's skip it.